python_transport/wirepas_gateway/transport_service.py

```diff
@@ -238,6 +324,15 @@ def __init__(self, settings, logger=None, **kwargs):
         else:
             self.data_event_id = None
 
+        self.net_monitoring_thread = NetMonitoringThread(
```

Comment:
Should be started only if WM_GW_BUFFERING_MONITOR_PERIOD is different than 0

Reply:
I would say that we have two options:

1) We keep the same pattern as the monitoring_thread: we avoid the object creation if not needed, and we check if the object is not None at each call site.
The transport_service has a dependancy on how the configuration is done, but we save some memory.

2) Or we always create the object. Then avoiding to start the Thread makes little sense because we are still adding a dependancy on how the configuration is done, but for no measurable gain.


I'm ok with both. But I don't see the in-between solution as an improvment.
Let me know.